content/en/docs/concepts/services-networking/dual-stack.md

```diff
@@ -68,47 +72,162 @@ An example of an IPv6 CIDR: `fdXY:IJKL:MNOP:15::/64` (this shows the format but
-   * `IPv6`: The API server will assign an IP from a `service-cluster-ip-range` that is `ipv6`
+The first family you list is used for the legacy `.spec.ClusterIP` field.
 
-The following Service specification does not include the `ipFamily` field. Kubernetes will assign an IP address (also known as a "cluster IP") from the first configured `service-cluster-ip-range` to this Service.
+### Dual-stack Service configuration scenarios
+
+These examples demonstrate the behavior of various dual-stack Service configuration scenarios.
+
+#### Dual-stack options on new Services
+
+   1. This Service specification does not explicitly define `.spec.ipFamilyPolicy`. When you create this Service, Kubernetes assigns a cluster IP for the Service from the first configured `service-cluster-ip-range` and sets the `.spec.ipFamilyPolicy` to `SingleStack`. ([Services without selectors](/docs/concepts/services-networking/service/#services-without-selectors) and [headless Services](/docs/concepts/services-networking/service/#headless-services) with selectors will behave in this same way.)
 
 {{< codenew file="service/networking/dual-stack-default-svc.yaml" >}}
 
-The following Service specification includes the `ipFamily` field. Kubernetes will assign an IPv6 address (also known as a "cluster IP") from the configured `service-cluster-ip-range` to this Service.
+   2. This Service specification explicitly defines `PreferDualStack` in `.spec.ipFamilyPolicy`. When you create this Service on a dual-stack cluster, Kubernetes assigns both IPv4 and IPv6 addresses for the service. The control plane updates the `.spec` for the Service to record the IP address assignments. The field `.spec.ClusterIPs` is the primary field, and contains both assigned IP addresses; `.spec.ClusterIP` is a secondary field with its value calculated from `.spec.ClusterIPs`. 
+   
+      * For the `.spec.ClusterIP` field, the control plane records the IP address that is from the same address family as the first service cluster IP range. 
+      * On a single-stack cluster, the `.spec.ClusterIPs` and `.spec.ClusterIP` fields both only list one address. 
+      * On a cluster with dual-stack enabled, specifying `RequireDualStack` in `.spec.ipFamilyPolicy` behaves the same as `PreferDualStack`.
 
-{{< codenew file="service/networking/dual-stack-ipv6-svc.yaml" >}}
+{{< codenew file="service/networking/dual-stack-preferred-svc.yaml" >}}
 
-For comparison, the following Service specification will be assigned an IPv4 address (also known as a "cluster IP") from the configured `service-cluster-ip-range` to this Service.
+   3. This Service specification explicitly defines `IPv6` and `IPv4` in `.spec.ipFamilies` as well as defining `PreferDualStack` in `.spec.ipFamilyPolicy`. Kubernetes will assign an IPv6 and IPv4 address in `.spec.ClusterIPs` and `.spec.ClusterIP` will be set to the IPv6 address because it is the first element in the `.spec.ClusterIPs` array, overriding the default. 
```

Comment:
What do you think about this edit:
When Kubernetes assigns an IPv6 and IPv4 address in `.spec.ClusterIPs`, the `.spec.ClusterIP` field is set to the IPv6 address because it is the first element in the `.spec.ClusterIPs` array, overriding the default.